File: examples/typescript/evm/getAccount.ts
// Usage: pnpm tsx evm/getAccount.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const cdp = new CdpClient();

let account = await cdp.evm.createAccount({
  name: `Account-${Math.floor(Math.random() * 100)}`,
});
console.log("Created account:", account.address);

account = await cdp.evm.getAccount({ address: account.address });
console.log("Retrieved account by address:", account.address);

account = await cdp.evm.getAccount({ name: account.name });
console.log("Retrieved account by name:", account.name);

File: examples/python/evm/get_account.py
# Usage: uv run python evm/get_account.py

import asyncio

from cdp import CdpClient
from dotenv import load_dotenv

load_dotenv()


async def main():
    async with CdpClient() as cdp:
        account = await cdp.evm.create_account(name="Account1")
        account = await cdp.evm.get_account(address=account.address)
        print("Account Address: ", account.address)
        account = await cdp.evm.get_account(name=account.name)
        print("Account Name: ", account.name)


asyncio.run(main())
